test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/NodeFilterTests.cs

```diff
@@ -0,0 +1,196 @@
+            Assert.That(allFiltersValid, Is.EqualTo(expectedAllFilters), "ConcatProperties did not construct the URN filter string as expected");
+
+            string sql2016ServerVersion = INodeFilter.ConcatProperties(nodeList, typeof(SqlHistoryTableQuerier), ValidForFlag.Sql2016);
+            string expectedSql2016Filters = "[((@TemporalType = 1)) and (@IsSystemObject = 1)]";
+            Assert.That(sql2016ServerVersion, Is.EqualTo(expectedSql2016Filters),  "ConcatProperties did not construct the URN filter string as expected when excluding filters that aren't valid for the given server type.");
+
+            string invalidQuerierType = INodeFilter.ConcatProperties(nodeList, typeof(SqlTableQuerier), ValidForFlag.Sql2022);
+            string expectedTableQuerierFilters = "[(@IsSystemObject = 1)]";
+            Assert.That(invalidQuerierType, Is.EqualTo(expectedTableQuerierFilters), "ConcatProperties did not construct the URN filter string as expected when excluding filters that don't match the querier type.");
+        }
+
+        /// <summary>
+        /// Validates the output of ConcatProperties with a list of multiple NodeOrFilters with more than 2 filters, and
+        /// a lone NodePropertyFilter
+        /// </summary>
+        [Test]
+        public void ConcatPropertiesWithMixedFilters()
+        {
+            // All these filters together are nonsense, but it's just testing the logic for constructing the filter string
+            var orNode = new NodeOrFilter {
+                FilterList = new List<NodePropertyFilter> {
+                    TemporalFilter,
+                    LedgerHistoryFilter
+                }
+            };
+
+            var orNode2 = new NodeOrFilter {
+                FilterList = new List<NodePropertyFilter> {
+                    SystemObjectFilter,
+                    LedgerHistoryFilter,
+                    TemporalFilter
+                }
+            };
+
+            var nodeList = new List<INodeFilter> {
+                orNode,
+                SystemObjectFilter,
+                orNode2
+            };
+
+            string allFiltersValid = INodeFilter.ConcatProperties(nodeList, typeof(SqlHistoryTableQuerier), ValidForFlag.Sql2022);
+            string expectedAllFilters = "[((@TemporalType = 1) or (@LedgerType = 1)) and (@IsSystemObject = 1) and ((@IsSystemObject = 1) or (@LedgerType = 1) or (@TemporalType = 1))]";
+            Assert.That(allFiltersValid, Is.EqualTo(expectedAllFilters), "ConcatProperties did not construct the URN filter string as expected");
+
+            string sql2016ServerVersion = INodeFilter.ConcatProperties(nodeList, typeof(SqlHistoryTableQuerier), ValidForFlag.Sql2016);
+            string expectedSql2016Filters = "[((@TemporalType = 1)) and (@IsSystemObject = 1) and ((@IsSystemObject = 1) or (@TemporalType = 1))]";
+            Assert.AreEqual(expectedSql2016Filters, sql2016ServerVersion);
```

Comment:
`Assert.That(...)`

Reply:
Yes, that was below. Just missed deleting the AreEqual.